vsphere/internal/virtualdevice/virtual_machine_disk_subresource.go

```diff
@@ -1253,6 +1266,27 @@ func (r *DiskSubresource) Read(l object.VirtualDeviceList) error {
 			r.Set("io_share_count", shares.Shares)
 		}
 	}
+
+	// Set storage policy if either it is template VM  with clone going on
+	// or VM already exists with update going on.
+	vmUUID := r.rdd.Get("clone.0.template_uuid").(string)
+	if vmUUID == "" {
+		// VM is not template. Check for UUID.
+		vmUUID = r.rdd.Get("uuid").(string)
+	}
+	if vmUUID != "" {
+		vm, err := virtualmachine.FromUUID(r.client, vmUUID)
+		if err != nil {
+			return err
+		}
+		vmMOID := vm.Reference().Value
```

Comment:
We've got a convenience function to simplify getting the MOID from the UUID you can use [here](https://github.com/terraform-providers/terraform-provider-vsphere/blob/master/vsphere/internal/helper/virtualmachine/virtual_machine_helper.go#L622).